MinitexPDFProtocols/MinitexPDFProtocols.swift

```diff
@@ -9,67 +9,92 @@
 import Foundation
 import UIKit
 
-// the items that are marked required are needed to
-// recreate an annotation in PSPDFKit later
-@objc public class MinitexPDFAnnotation: NSObject, Codable {
-  // required
-  public let pageIndex: UInt
-  // required
-  public let type: String
-  // required
-  public let bbox: String
-  // required
-  public let rects: [String]
 
-  // optional
-  public let color: String?
-  // optional
-  public let opacity: Float?
-  // optional
-  // we store this in case the field parsing fails, then we have the original JSON
-  public let JSONData: Data?
+/// A generic annotation in a PDF document.
+@objc public protocol MinitexPDFAnnotation: class {
 
-  // optional
-  // this value cannot be set in PSPDFKit
-  //var v: Int?
+  /// An annotation must belong to one page of the PDF document.
+  var page: MinitexPDFPage { get }
 
-  public init(pageIndex: UInt, type: String, bbox: String, rects: [String],
-              color: String?, opacity: Float?, JSONData: Data?) {
-    self.pageIndex = pageIndex
-    self.type = type
-    self.bbox = bbox
-    self.rects = rects
-    self.color = color
-    self.opacity = opacity
-    self.JSONData = JSONData
-  }
+  /// Complex annotations may be fully described by something like JSON from a particular
+  /// renderer. This allows easy access by the host to a version it can store or sync.
+  var serializedRepresentation: Data { get }
 }
 
-
+/// The View Controller for the PDF document.
 @objc public protocol MinitexPDFViewController: class {
-  // we have to pass in a dictionary because @objc protocol function
-  // cannot accept multiple parameters
-  @objc init(dictionary: [String: Any])
+
+  /// Initialize a view controller with a particular PDF document.
+  ///
+  /// - Parameters:
+  ///   - file: The local file directory of the PDF document.
+  ///   - page: An optional page to navigate to when first opening.
+  ///   - bookmarks: Any additional user-saved bookmarks.
+  ///   - annotations: Any additional user-created generic annotations.
+  init?(file: URL,
+        openToPage page: MinitexPDFPage?,
+        bookmarks: [MinitexPDFPage]?,
+        annotations: [MinitexPDFAnnotation]?)
 }
 
+/// A page in a PDF document.
+@objc public protocol MinitexPDFPage: class {
+
+  /// A PDF document must have an unsigned integer represenation of its pages.
+  var pageNumber: UInt { get }
+}
 
 @objc public protocol MinitexPDFViewControllerDelegate: class {
-  @objc func userDidNavigate(page: Int)
-  @objc func saveBookmarks(pageNumbers: [UInt])
-  @objc func saveAnnotations(annotations: [MinitexPDFAnnotation])
```

Comment:
It's up to the API consumer / host to decide if it wants to "save" bookmarks or not. Apple's naming pattern on delegate methods is exactly like the first method "userDidNavigate", so all the methods names are updated for consistency.
As long as a delegate receives every bookmark created, it can keep its own array of them saved if it wants to. So it's just sticking to "delegation" as "not making decisions for the host"

Documentation is added as well.